File: src/main/java/ru/abe/slaves/potrebot/web/model/VkMessage.java
package ru.abe.slaves.potrebot.web.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.Date;

@Data
public class VkMessage {
    public static final int CHAT_PEER_PREFIX = 2000000000;

    private Date date;
    @JsonProperty("from_id")
    private int fromId;
    private int id;
    private int out;
    @JsonProperty("peer_id")
    private int peerId;
    private String text;

    public int getChatId() {
        return peerId - CHAT_PEER_PREFIX;
    }
}

File: src/main/java/ru/abe/slaves/potrebot/web/model/VkMessage.kt
package ru.abe.slaves.potrebot.web.model

import com.google.gson.annotations.SerializedName

private const val CHAT_PEER_PREFIX = 2000000000

data class VkMessage(
    @SerializedName("from_id")
    val fromId: Int,
    val id: Int,
    val out: Int,
    @SerializedName("peer_id")
    val peerId: Int,
    val text: String
) {
    val chatId: Int
        get() = peerId - CHAT_PEER_PREFIX
}